services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/TransactUpdateItemEnhancedRequest.java

```diff
@@ -0,0 +1,239 @@
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.enhanced.dynamodb.model;
+
+import java.util.Objects;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbEnhancedAsyncClient;
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbEnhancedClient;
+import software.amazon.awssdk.enhanced.dynamodb.Expression;
+import software.amazon.awssdk.services.dynamodb.model.ReturnValuesOnConditionCheckFailure;
+
+/**
+ * Defines parameters used to update an item to a DynamoDb table using the
+ * {@link DynamoDbEnhancedClient#transactWriteItems(TransactWriteItemsEnhancedRequest)} or
+ * {@link DynamoDbEnhancedAsyncClient#transactWriteItems(TransactWriteItemsEnhancedRequest)}
+ * operation.
+ * <p>
+ * A valid request object must contain the item that should be written to the table.
+ *
+ * @param <T> The type of the modelled object.
+ */
+@SdkPublicApi
+public class TransactUpdateItemEnhancedRequest<T> {
+
+    private final T item;
+    private final Boolean ignoreNulls;
+    private final Expression conditionExpression;
+    private final String returnValuesOnConditionCheckFailure;
+
+    private TransactUpdateItemEnhancedRequest(Builder<T> builder) {
+        this.item = builder.item;
+        this.ignoreNulls = builder.ignoreNulls;
+        this.conditionExpression = builder.conditionExpression;
+        this.returnValuesOnConditionCheckFailure = builder.returnValuesOnConditionCheckFailure;
+    }
+
+    /**
+     * Creates a newly initialized builder for the request object.
+     *
+     * @param itemClass the class that items in this table map to
+     * @param <T>       The type of the modelled object, corresponding to itemClass
+     * @return a UpdateItemEnhancedRequest builder
+     */
+    public static <T> Builder<T> builder(Class<? extends T> itemClass) {
+        return new Builder<>();
+    }
+
+    /**
+     * Returns a builder initialized with all existing values on the request object.
+     */
+    public Builder<T> toBuilder() {
+        return new Builder<T>().item(item)
+                               .ignoreNulls(ignoreNulls)
+                               .conditionExpression(conditionExpression)
+                               .returnValuesOnConditionCheckFailure(returnValuesOnConditionCheckFailure);
+    }
+
+    /**
+     * Returns the item for this update operation request.
+     */
+    public T item() {
+        return item;
+    }
+
+    /**
+     * Returns if the update operation should ignore attributes with null values, or false if it has not been set.
+     */
+    public Boolean ignoreNulls() {
```

Comment:
Copy+paste from the normal `UpdateItemEnhancedRequest.java` :). But in general, for modeled request objects we always use the nullable version because null may not mean the same as false.